modal/_functions.py

```diff
@@ -256,45 +258,50 @@ async def _retry_input(self) -> None:
             request,
         )
 
-    async def _get_single_output(self, expected_jwt: Optional[str] = None) -> Any:
+    async def _get_single_output_wrap_exceptions(self, expected_jwt: Optional[str] = None) -> Any:
         # waits indefinitely for a single result for the function, and clear the outputs buffer after
+        # raises wrapped internal _ContainerExceptions if they occur
         item: api_pb2.FunctionGetOutputsItem = (
             await self.pop_function_call_outputs(
                 timeout=None,
                 clear_on_success=True,
                 input_jwts=[expected_jwt] if expected_jwt else None,
             )
         ).outputs[0]
-        return await _process_result(item.result, item.data_format, self.stub, self.client)
+        return await _process_result_wrap_exceptions(item.result, item.data_format, self.stub, self.client)
 
     async def run_function(self) -> Any:
-        # Use retry logic only if retry policy is specified and
-        ctx = self._retry_context
-        if (
-            not ctx
-            or not ctx.retry_policy
-            or ctx.retry_policy.retries == 0
-            or ctx.function_call_invocation_type != api_pb2.FUNCTION_CALL_INVOCATION_TYPE_SYNC
-            or not ctx.sync_client_retries_enabled
-        ):
-            return await self._get_single_output()
+        try:
```

Comment:
This is the only place where we actually "need" the exception wrapping (in order to detect retryable errors), so it's explicitly using `_get_single_output_wrap_exceptions` and then unwrapping it in an outer try/except